lib/modifications.rb

```diff
@@ -29,74 +35,58 @@ def self.modify!(results)
         bundle.entry.first.resource.address.first.period.start = bundle.entry.first.resource.birthDate
       end
 
-      # Add discharge disposition to first encounter of each record
+      # Make sure there aren't stupid numbers of every resource type
+      missing_profiles = DataScript::Constraints::REQUIRED_PROFILES.dup
       results.each do |bundle|
-        encounter_entry = bundle.entry.find {|e| e.resource.resourceType == 'Encounter'}
-        encounter = encounter_entry.resource
-        encounter.hospitalization = FHIR::Encounter::Hospitalization.new
-        encounter.hospitalization.dischargeDisposition = create_codeable_concept('http://www.nubc.org/patient-discharge','01','Discharged to home care or self care (routine discharge)')
+        resource_counts = get_resource_counts(bundle).sort
+        # Move DocumentReferences to the front, so we delete them first (and don't have reference issues)
+        resource_counts.insert(0, resource_counts.delete(resource_counts.find { |k, _| k == 'DocumentReference' }))
+        # Move Encounters to the end, so we know which ones are safe to delete
+        resource_counts.append(resource_counts.delete(resource_counts.find { |k, _| k == 'Encounter' }))
+        provenance = bundle.entry.find { |e| e.resource.is_a? FHIR::Provenance }.resource
+        resource_counts.each do |type, count|
+          next unless count >= 30
```

Comment:
Is there a reason 30 is the magic number? Does this need to be configurable?

Reply:
At the moment 30 is really just a magic number; no real significance to it other than it's relatively low.